xarray/core/datatree.py

```diff
@@ -429,71 +527,85 @@ def parent(self: DataTree, new_parent: DataTree) -> None:
             raise ValueError("Cannot set an unnamed node as a child of another node")
         self._set_parent(new_parent, self.name)
 
+    def _to_dataset_view(self, rebuild_dims: bool) -> DatasetView:
+        variables = dict(self._data_variables)
+        variables |= self._coord_variables
```

Comment:
Couldn't you just do this?
```suggestion
        variables = dict(self._data_variables) | self._coord_variables
```

Reply:
This would work, but would involve an extra `dict` copy.

`self._data_variables | self._coord_variables` also works (it's what I had originally), but Mypy contains about a (possibly mistaken?) type error.